superset/datasets/columns/api.py

```diff
@@ -36,14 +38,99 @@
 class DatasetColumnsRestApi(BaseSupersetModelRestApi):
     datamodel = SQLAInterface(TableColumn)
 
-    include_route_methods = {"delete"}
+    allow_browser_login = True
     class_permission_name = "Dataset"
+    include_route_methods = {RouteMethod.DELETE, RouteMethod.GET, RouteMethod.GET_LIST}
     method_permission_name = MODEL_API_RW_METHOD_PERMISSION_MAP
-
+    list_columns = [  # See DatasetRestApi.show_select_columns.
```

Comment:
I'm not overly sure the difference between `list_columns` and `show_columns` but I thought that it was prudent that the `table_columns` columns returned was consistent with the columns field in the `/api/v1/dataset` response.

Reply:
`list_columns` is for get_list and `show_columns` is for `get` item. Not great naming, at the time I aimed for compatibility with the previous `MovelView` classes on FAB to allow for a smooth migration

Reply:
Can we use this opportunity to add comments in the code explaining the difference?